src/facade/cmd_arg_parser.h

```diff
@@ -65,32 +65,46 @@ struct CmdArgParser {
   void ExpectTag(std::string_view tag);
 
   // Consume next value
-  template <class... Cases> auto Switch(Cases&&... cases) {
+  template <class... Cases> auto Map(Cases&&... cases) {
     if (cur_i_ >= args_.size()) {
       Report(OUT_OF_BOUNDS, cur_i_);
-      return typename decltype(SwitchImpl(std::string_view(),
-                                          std::forward<Cases>(cases)...))::value_type{};
+      return typename decltype(MapImpl(std::string_view(),
+                                       std::forward<Cases>(cases)...))::value_type{};
     }
 
     auto idx = cur_i_++;
-    auto res = SwitchImpl(SafeSV(idx), std::forward<Cases>(cases)...);
+    auto res = MapImpl(SafeSV(idx), std::forward<Cases>(cases)...);
     if (!res) {
       Report(INVALID_CASES, idx);
       return typename decltype(res)::value_type{};
     }
     return *res;
   }
 
-  // Check if the next value if equal to a specific tag. If equal, its consumed.
-  bool Check(std::string_view tag) {
-    if (cur_i_ >= args_.size())
+  // Consume next value if can map it and return mapped result or return nullopt
+  template <class... Cases>
+  auto CheckMap(Cases&&... cases) -> std::optional<std::tuple_element_t<1, std::tuple<Cases...>>> {
```

Comment:
names are quite confusing. Map vs CheckMap
When you call Map you will fail if one of the cases does not exist
but in CheckMap you will not fail. Is this right? is this the only difference between funcitons?

Reply:
yes

Reply:
How about MapNext (must map to one of the options) and TryMapNext (return nullopt if non option apply)

Reply:
I thought about it but decided to be consistent with the Check method if you think that it will be better I will apply it

Reply:
I think the names are very confusing. I understand how it continues the Check logic but maybe the Check name should be refactored but not in this PR

Reply:
ok